### PR TITLE
Replace pylint `useless-assign` with ruff `RET504`

### PR DESCRIPTION
**Pull request by harupy**
<details><summary>&#x1F6E0 DevTools &#x1F6E0</summary>
<p>

[Open in GitHub Codespaces](https://codespaces.new/harupy/mlflow/pull/11151?quickstart=1)

#### Install mlflow from this PR

```
pip install git+https://github.com/mlflow/mlflow.git@refs/pull/11151/merge
```

#### Checkout with GitHub CLI

```
gh pr checkout 11151
```

</p>
</details>

### Related Issues/PRs

 #xxx

### What changes are proposed in this pull request?

Replace pylint `useless-assign` (custom rule) with ruff `RET504`.

### How is this PR tested?

- [x] Existing unit/integration tests
- [ ] New unit/integration tests
- [ ] Manual tests

### Does this PR require documentation update?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

#### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Review by B-Step62 — COMMENTED** [review]
Tbh I'm not too sure if we want to apply this rule to all notebooks/examples. Sometimes it feels naming the return explicitly makes the intent clearer. 

For example,
```
transcript_section = soup.find(\"div\", class_=div_class)
if transcript_section:
    transcript_text = transcript_section.get_text(separator=\"\\n\", strip=True)
    return transcript_text
```

```
transcript_section = soup.find(\"div\", class_=div_class)
if transcript_section:
    return transcript_section.get_text(separator=\"\\n\", strip=True)
```

It's more easy to tell the return is transcript text.

**Comment by harupy** [comment]
Fair enough. We can exclude notebooks/examples.

File: docs/source/llms/langchain/notebooks/langchain-retriever.ipynb [hunk 1]
```diff
@@ -117,8 +117,7 @@
     "        # Finding the transcript section by its HTML structure\n",
     "        transcript_section = soup.find(\"div\", class_=div_class)\n",
     "        if transcript_section:\n",
-    "            transcript_text = transcript_section.get_text(separator=\"\\n\", strip=True)\n",
-    "            return transcript_text\n",
+    "            return transcript_section.get_text(separator=\"\\n\", strip=True)\n",
```
Comment (harupy) [inline docs/source/llms/langchain/notebooks/langchain-retriever.ipynb:120]:
```suggestion
    "            transcript_text = transcript_section.get_text(separator=\"\\n\", strip=True)\n",
    "            return transcript_text\n",```

File: docs/source/llms/langchain/notebooks/langchain-retriever.ipynb [hunk 2]
```diff
@@ -118,7 +118,7 @@
     "        transcript_section = soup.find(\"div\", class_=div_class)\n",
     "        if transcript_section:\n",
     "            transcript_text = transcript_section.get_text(separator=\"\\n\", strip=True)\n",
-    "            return transcript_text\n",
+    "            return transcript_text\n",```
```
Comment (harupy) [inline docs/source/llms/langchain/notebooks/langchain-retriever.ipynb:121]:
```suggestion
    "            return transcript_text\n",
```

**Comment by harupy** [comment]
@B-Step62 could you take another look?

File: examples/pytorch/AxHyperOptimizationPTL/iris.py [hunk 3]
```diff
@@ -34,8 +34,7 @@ def __init__(self, **kwargs):
     def forward(self, x):
         x = F.relu(self.fc1(x))
         x = F.relu(self.fc2(x))
-        x = F.relu(self.fc3(x))
-        return x
+        return F.relu(self.fc3(x))
```
Comment (B-Step62) [inline examples/pytorch/AxHyperOptimizationPTL/iris.py:38]: I hardly saw ppl doing this, rather than vertically aligning layers like the original one...

**Review by B-Step62 — APPROVED** [review]
LGTM, while I'd prefer keeping original style for those notebooks defining model layars...vertically stucking layers looks clearner to me than putting the last layer alone in return statement. That said after all it's just a personal preference so feel free to ignore if you like  the new one more:)

File: examples/pytorch/AxHyperOptimizationPTL/iris.py (hunk 3 above)
Comment (harupy) [reply examples/pytorch/AxHyperOptimizationPTL/iris.py:38]: Ok, then let's revert them :)

File: examples/pytorch/AxHyperOptimizationPTL/iris.py [hunk 4]
```diff
@@ -34,8 +34,7 @@ def __init__(self, **kwargs):
...
```
Comment (harupy) [inline examples/pytorch/AxHyperOptimizationPTL/iris.py:37]:
```suggestion
        x = F.relu(self.fc3(x))
        return x
```

File: examples/pytorch/torchscript/IrisClassification/iris_classification.py [hunk 5]
```diff
@@ -23,8 +23,7 @@ def forward(self, x):
         x = F.relu(self.fc1(x))
         x = F.relu(self.fc2(x))
         x = F.dropout(x, 0.2)
-        x = self.fc3(x)
-        return x
+        return self.fc3(x)
```
Comment (harupy) [inline examples/pytorch/torchscript/IrisClassification/iris_classification.py:26]:
```suggestion
        x = self.fc3(x)
        return x
```

File: examples/pytorch/torchscript/MNIST/mnist_torchscript.py [hunk 6]
```diff
@@ -34,8 +34,7 @@ def forward(self, x):
         x = F.relu(x)
         x = self.dropout2(x)
         x = self.fc2(x)
-        output = F.log_softmax(x, dim=1)
-        return output
+        return F.log_softmax(x, dim=1)
```
Comment (harupy) [inline examples/pytorch/torchscript/MNIST/mnist_torchscript.py:37]:
```suggestion
        x = F.log_softmax(x, dim=1)
```

File: examples/pytorch/BertNewsClassification/bert_classification.py [hunk 7]
```diff
@@ -264,8 +264,7 @@ def forward(self, input_ids, attention_mask):
         output = self.bert_model(input_ids=input_ids, attention_mask=attention_mask)
         output = F.relu(self.fc1(output.pooler_output))
         output = self.drop(output)
-        output = self.out(output)
-        return output
+        return self.out(output)
```
Comment (harupy) [inline examples/pytorch/BertNewsClassification/bert_classification.py:267]:
```suggestion
        output = self.out(output)
        return output
```

File: examples/pytorch/IterativePruning/mnist.py [hunk 8]
```diff
@@ -128,9 +128,7 @@ def forward(self, x):
         x = self.layer_3(x)
 
         # probability distribution over labels
-        x = torch.log_softmax(x, dim=1)
-
-        return x
+        return torch.log_softmax(x, dim=1)
```
Comment (harupy) [inline examples/pytorch/IterativePruning/mnist.py:131]:
```suggestion
        x = torch.log_softmax(x, dim=1)

        return x
```

File: examples/pytorch/MNIST/mnist_autolog_example.py [hunk 9]
```diff
@@ -134,9 +134,7 @@ def forward(self, x):
         x = self.layer_3(x)
 
         # probability distribution over labels
-        x = torch.log_softmax(x, dim=1)
-
-        return x
+        return torch.log_softmax(x, dim=1)
```
Comment (harupy) [inline examples/pytorch/MNIST/mnist_autolog_example.py:137]:
```suggestion
        x = torch.log_softmax(x, dim=1)

        return x
```

File: examples/pytorch/logging/pytorch_log_model.ipynb [hunk 10]
```diff
@@ -201,8 +201,7 @@
     "\n",
     "    def forward(self, x):\n",
     "        x = torch.relu(self.fc1(x))\n",
-    "        x = self.fc2(x)\n",
-    "        return x\n",
+    "        return self.fc2(x)\n",
```
Comment (harupy) [inline examples/pytorch/logging/pytorch_log_model.ipynb:204]:
```suggestion
    "        x = self.fc2(x)\n",
    "        return x\n",
```

File: examples/pytorch/torchscript/MNIST/mnist_torchscript.py [hunk 11]
```diff
@@ -34,8 +34,7 @@ def forward(self, x):
         x = F.relu(x)
         x = self.dropout2(x)
         x = self.fc2(x)
-        output = F.log_softmax(x, dim=1)
-        return output
+        x = F.log_softmax(x, dim=1)
```
Comment (harupy) [inline examples/pytorch/torchscript/MNIST/mnist_torchscript.py:37]:
```suggestion
        x = F.log_softmax(x, dim=1)
        return x
```

File: examples/tensorflow/train.py [hunk 12]
```diff
@@ -56,7 +56,6 @@ def __call__(self, x):
         x = self.norm_x.norm(x)
         y = self.model(x)
         y = self.norm_y.unnorm(y)
```
Comment (harupy) [inline examples/tensorflow/train.py:58]:
```suggestion
        y = self.norm_y.unnorm(y)
        return y
```

File: examples/paddle/train_high_level_api.py [hunk 13]
```diff
@@ -13,8 +13,7 @@ def __init__(self):
         self.fc_ = paddle.nn.Linear(13, 1, None)
 
     def forward(self, inputs):
-        pred = self.fc_(inputs)
-        return pred
+        return self.fc_(inputs)
```
Comment (harupy) [inline examples/paddle/train_high_level_api.py:16]:
```suggestion
        pred self.fc_(inputs)
        return pred
```

File: examples/paddle/train_low_level_api.py [hunk 14]
```diff
@@ -33,8 +33,7 @@ def __init__(self):
 
     @paddle.jit.to_static
     def forward(self, inputs):
-        x = self.fc(inputs)
-        return x
+        return self.fc(inputs)
```
Comment (harupy) [inline examples/paddle/train_low_level_api.py:36]:
```suggestion
        x = self.fc(inputs)
        return x
```

File: examples/paddle/train_high_level_api.py [hunk 15]
```diff
@@ -13,7 +13,7 @@ def __init__(self):
         self.fc_ = paddle.nn.Linear(13, 1, None)
 
     def forward(self, inputs):
-        pred = self.fc_(inputs)
+        pred self.fc_(inputs)
```
Comment (harupy) [inline examples/paddle/train_high_level_api.py:16]:
```suggestion
        pred = self.fc_(inputs)
```

**Comment by harupy** [comment]
@mlflow-automation autoformat

File: mlflow/pyfunc/__init__.py [hunk 16]
```diff
@@ -2194,7 +2194,9 @@ def predict(self, context, model_input: List[str], params=None) -> List[str]:
 
 
                 with mlflow.start_run():
-                    model_info = mlflow.pyfunc.log_model(artifact_path="model", python_model=MyModel())  # noqa # pylint: disable=line-too-long
+                    model_info = mlflow.pyfunc.log_model(
+                        artifact_path="model", python_model=MyModel()
+                    )  # noqa # pylint: disable=line-too-long
```
Comment (harupy) [inline mlflow/pyfunc/__init__.py:2199]:
```suggest
                    model_info = mlflow.pyfunc.log_model(artifact_path="model", python_model=MyModel())  # noqa # pylint: disable=line-too-long
```

Comment (harupy) [inline mlflow/pyfunc/__init__.py:2199]:
```suggestion
                    model_info = mlflow.pyfunc.log_model(artifact_path="model", python_model=MyModel())  # noqa # pylint: disable=line-too-long
```

File: examples/pytorch/AxHyperOptimizationPTL/iris.py (hunk 3 above)
Comment (harupy) [reply examples/pytorch/AxHyperOptimizationPTL/iris.py]: @B-Step62 reverted
